gradle-plugin/src/test/kotlin/io/spine/tools/prototap/gradle/PluginSpec.kt

```diff
@@ -0,0 +1,145 @@
+import io.spine.tools.gradle.testing.GradleProject
+import io.spine.tools.prototap.Names.GRADLE_PLUGIN_ID
+import io.spine.tools.prototap.Paths.CODE_GENERATOR_REQUEST_FILE
+import io.spine.tools.prototap.Paths.DESCRIPTOR_SET_FILE
+import io.spine.tools.prototap.Paths.outputRoot
+import java.io.File
+import java.nio.file.Path
+import kotlin.io.path.exists
+import org.gradle.testkit.runner.internal.DefaultGradleRunner
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.io.TempDir
+
+@DisplayName("Gradle plugin should")
+internal class PluginSpec {
+
+    private lateinit var projectDir: File
+    private lateinit var project: GradleProject
+    private lateinit var resultDir: Path
+
+    @BeforeEach
+    fun prepareDir(@TempDir tempDir: File) {
+        projectDir = tempDir
+        val buildDir = "$projectDir/build"
+        resultDir = outputRoot(buildDir)
+    }
+
+    @Test
+    fun `run with default values`() {
+        createProject("default-values")
+        runBuild()
+        assertJavaCodeGenerated()
+        assertRequestFileExits()
+    }
+
+    @Test
+    fun `run with proto files under 'tests'`() {
+        createProject("proto-in-test")
+        runBuild()
+        assertJavaCodeGenerated()
+        assertRequestFileExits()
+    }
+
+    @Test
+    fun `run with 'prototap' settings`() {
+        createProject("with-settings")
+        runBuild()
+        assertJavaCodeGenerated()
+        assertRequestFileExits()
+        assertDescriptorSetFileExits()
+    }
+
+    @Test
+    fun `work as a 'classpath' dependency of 'buildscript'`() {
+        createProject("via-classpath")
+        runBuild()
+        assertJavaCodeGenerated()
+        assertRequestFileExits()
+    }
+
+    private fun assertJavaCodeGenerated() {
+        val javaDir = resultDir.resolve("java")
+        javaDir.countFiles() shouldNotBe 0
+        val packageDir = javaDir.resolve("io/spine/given/domain/gas/")
+        packageDir.run {
+            exists() shouldBe true
+            // Check some Java files too.
+            resolve("Pipe.java").exists() shouldBe true
+            resolve("CompressorStation.java").exists() shouldBe true
+            resolve("LngStorage.java").exists() shouldBe true
+        }
+    }
+
+    private fun assertRequestFileExits() {
+        resultDir.resolve(CODE_GENERATOR_REQUEST_FILE).exists() shouldBe true
+    }
+
+    private fun assertDescriptorSetFileExits() {
+        resultDir.resolve(DESCRIPTOR_SET_FILE).exists() shouldBe true
+    }
+
+    private fun createProject(resourceDir: String) {
+        val version = Plugin.readVersion()
+        val builder = GradleProject.setupAt(projectDir)
+            .fromResources(resourceDir)
+            .withSharedTestKitDirectory()
+            .replace("@PROTOTAP_PLUGIN_ID@", GRADLE_PLUGIN_ID)
+            .replace("@PROTOTAP_VERSION@", version)
+            //.withLoggingLevel(LogLevel.INFO)
```

Comment:
I would move this line under `//.enableRunnerDebug()`, because it also makes sense when investigating some issue, such as a test failure.